### PR TITLE
The Great Awaitening

### PR DESCRIPTION
**Pull request by chris-olszewski**
The realization of #246, should be merged after #248

### CODE REVIEW

**Comment by dtetto** [comment]
Very good PR title 👍 👍

File: src/util/load.js [hunk 1]
```diff
@@ -43,33 +42,36 @@ export function events (func, eventName) {
...
+    const api = await readJSON('api.json')
+    return api
   } catch (e) {
     return null
   }
 }
 
-export function babelrc () {
-  return readJSONSync('.babelrc')
+export async function babelrc () {
```
Comment (chris-olszewski) [inline src/util/load.js]: @reconbot Is it good/bad practice to mark functions that return a promise as `async` even if they don't use `await` in the body?

File: src/util/remove-environment.js [hunk 2]
```diff
@@ -6,7 +6,5 @@ const pattern = '*'
 
 export default async function (env, vars) {
   const fns = await funcs(pattern)
-  return Promise.all(fns.map(async (func) => {
-    return removeEnvVars(env, lambdaConfig(func), vars)
-  }))
+  return Promise.map(fns, async (func) => removeEnvVars(env, await lambdaConfig(func), vars))
```
Comment (chris-olszewski) [inline src/util/remove-environment.js]: Better to have
```
const configs = await Promise.map(funcs(pattern), lambdaConfig)
return Promise.map(configs, (config) => removeEnvVars(env, config, vars))
```

File: src/util/upload-environment.js [hunk 3]
```diff
@@ -5,8 +5,6 @@ import { lambdaConfig, funcs } from './load'
 const pattern = '*'
 
 export default async function (env, vars) {
-  const fns = await funcs(pattern)
-  return Promise.map(fns, async (func) => {
-    return putEnvironment(env, lambdaConfig(func), vars)
-  })
+  const configs = await Promise.map(funcs(pattern), lambdaConfig)
+  return Promise.map(configs, (config) => putEnvironment(env, config, vars), { concurrency: 3 })
```
Comment (chris-olszewski) [inline src/util/upload-environment.js]: change `concurrency` to 2, 3 seemed to break stuff

**Pull request merged**